core/store/src/config.rs

```diff
@@ -174,6 +189,31 @@ impl StoreConfig {
             _ => bytesize::ByteSize::mib(32),
         }
     }
+
+    fn default_per_shard_max_bytes() -> HashMap<ShardUId, bytesize::ByteSize> {
+        let epoch_config_store = EpochConfigStore::for_chain_id(MAINNET, None).unwrap();
+        let mut shard_layouts: Vec<ShardLayout> = Vec::new();
+        // 72 is the latest protocol version of config store as of writing
+        for protocol_version in 72..=PROTOCOL_VERSION {
+            let epoch_config = epoch_config_store.get_config(protocol_version);
+            let shard_layout = epoch_config.shard_layout.clone();
+            // O(n) is fine as list is short
+            if !shard_layouts.contains(&shard_layout) {
+                shard_layouts.push(shard_layout);
+            }
+        }
+
+        let mut per_shard_max_bytes: HashMap<ShardUId, bytesize::ByteSize> = HashMap::new();
+        for (shard_uid, bytes) in PER_ACCOUNT_CACHE_SIZE.iter() {
+            let account_id = AccountId::from_str(shard_uid)
+                .expect("the hardcoded account id should guarantee to be valid");
+            for shard_layout in shard_layouts.iter() {
+                let shard_uid = account_id_to_shard_uid(&account_id, &shard_layout);
+                per_shard_max_bytes.insert(shard_uid, *bytes);
+            }
+        }
+        per_shard_max_bytes
```

Comment:
I think we may still need to include the values from before protocol version 72 here from the original list?
```
                    // Temporary solution to make contracts with heavy trie access
                    // patterns on shard 3 more stable. It was chosen by the estimation
                    // of the largest contract storage size we are aware as of 23/08/2022.
                    // Note: on >= 1.34 nearcore version use 1gb if you have minimal hardware.
                    // In simple nightshade the heavy contract "token.sweat" is in shard 3
                    (ShardUId { version: 1, shard_id: 3 }, bytesize::ByteSize::gb(3)),
                    // In simple nightshade v2 the heavy contract "token.sweat" is in shard 4
                    (ShardUId { version: 2, shard_id: 4 }, bytesize::ByteSize::gb(3)),
                    // In simple nightshade v3 the heavy contract "token.sweat" is in shard 5
                    (ShardUId { version: 3, shard_id: 5 }, bytesize::ByteSize::gb(3)),
                    // Shard 1 is dedicated to aurora and it had very few cache
                    // misses even with cache size of only 50MB
                    (ShardUId { version: 1, shard_id: 1 }, bytesize::ByteSize::mb(50)),
                    (ShardUId { version: 2, shard_id: 1 }, bytesize::ByteSize::mb(50)),
                    (ShardUId { version: 3, shard_id: 1 }, bytesize::ByteSize::mb(50)),
```
Could you please quickly check if shard_uids with version 1, 2, 3 are a part of the list?

Reply:
Today we should only need version 3, that's the current prod shard layout.